### PR TITLE
USB: fix status callback in composite configuration and crash in wpanusb sample

### LINKED ISSUE #14882: USB DFU never enters DFU mode, when composite device is enabled and mcuboot is used
**Describe the bug**
When trying to use USB DFU in a composite device, the device never enters DFU mode. When you try to download new firmware using dfu-util, it fails with the message `Device still in Runtime Mode`

**To Reproduce**

- Add `CONFIG_USB_COMPOSITE_DEVICE=y` to config of the USB DFU sample.
- Flash mcuboot and the sample to the device
- Attempt a DFU using `sudo dfu-util --alt 1 --download signed-zephyr.bin --i 0`

**Impact**
We are unable to use USB DFU in our product the relies on USB communication.

**Screenshots or console output**
dfu-util output:
```
dfu-util 0.9

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2019 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

dfu-util: Invalid DFU suffix signature
dfu-util: A valid DFU suffix will be required in a future dfu-util release!!!
Opening DFU capable USB device...
ID 2fe3:0100
Run-time device DFU version 0110
Claiming USB DFU Runtime Interface...
Determining device status: state = appIDLE, status = 0
Device really in Runtime Mode, send DFU detach request...
Resetting USB...
Opening DFU USB Device...
Claiming USB DFU Interface...
Setting Alternate Setting #1 ...
Determining device status: state = appDETACH, status = 0
dfu-util: Device still in Runtime Mode!
```
Zephyr output:
```
***** Booting Zephyr OS v1.14.0-rc1-1394-g972221423cde *****
[00:00:00.000,396] <inf> main: This device supports USB DFU class.

[00:00:03.748,168] <dbg> usb_dfu.dfu_custom_handle_req: DFU alternate setting 0
[00:00:03.748,352] <dbg> usb_dfu.dfu_class_handle_req: DFU_GETSTATUS: status 0, state 0
[00:00:04.004,852] <dbg> usb_dfu.dfu_class_handle_req: DFU_DETACH timeout 1000, state 0
[00:00:04.563,629] <dbg> usb_dfu.dfu_status_cb: USB device configured
[00:00:09.286,804] <dbg> usb_dfu.dfu_custom_handle_req: DFU alternate setting 1
[00:00:09.287,384] <dbg> usb_dfu.dfu_class_handle_req: DFU_GETSTATUS: status 0, state 1
[00:00:09.543,975] <dbg> usb_dfu.dfu_custom_handle_req: DFU alternate setting 0
```
**Environment (please complete the following information):**
 - OS: Linux (Ubuntu)
 - Toolchain: Zephyr SDK
 - Commit SHA: 972221423cde92639f685542b10a8caa35793ec5

**Additional context**
I'm am unsure wheter or not this is related to this issue, but if also enable CDC ACM in the config, the DFU usb interface is listed as 2 instead of 0, and dfu-util fails with  `dfu-util: Lost device after RESET?` instead. After the first failed attempt, the interface number is 0 again and the issue is the same as above.

### PR DESCRIPTION
**Pull request by finikorg**
Fixes wpanusb, use usb_transfer()
Fixes #14882

### CODE REVIEW

File: subsys/usb/class/usb_dfu.c [hunk 1]
```diff
@@ -592,9 +592,12 @@ static int dfu_class_handle_req(struct usb_setup_packet *pSetup,
  *
  * @return  N/A.
  */
-static void dfu_status_cb(enum usb_dc_status_code status, const u8_t *param)
+static void dfu_status_cb(struct usb_cfg_data *cfg,
```
Comment (jfischer-phytec-iot) [inline subsys/usb/class/usb_dfu.c]: how I see it is alternative to #14960, then please add fixes: #14882 to commit message

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: subsys/usb/class/usb_dfu.c (hunk 1 above)
Comment (finikorg) [reply subsys/usb/class/usb_dfu.c]: That PR ( #14960) is a workaround. I am fine with it also before we merge the proper solution.

**Review by jfischer-no — APPROVED** [review]

**Comment by carlescufi** [comment]
@Qbicz @pdunaj could you please test this on Nordic hardware?

**Comment by pdunaj** [comment]
Hi @carlescufi , sure. We are on it.

**Comment by Qbicz** [comment]
Thank you for this work @finikorg. I tested it with USB DFU sample with `CONFIG_USB_COMPOSITE_DEVICE=y` on nRF52840. Unfortunately the outcome I see is the same as with https://github.com/zephyrproject-rtos/zephyr/pull/14960 - device is lost after reset request.

I attached verbose logs on device and dfu-util side.
[usb_device_composite_dfu.log](https://github.com/zephyrproject-rtos/zephyr/files/3039003/usb_device_composite_dfu.log)
[usb_dfu_util_composite.log](https://github.com/zephyrproject-rtos/zephyr/files/3039016/usb_dfu_util_composite.log)

**Comment by Qbicz** [comment]
@pawelzadrozniak FYI

**Review by carlescufi — APPROVED** [review]

**Pull request merged**